### PR TITLE
feature/OO-8570 - Chart Legend Status

### PR DESCRIPTION
**Pull request by pilat-martin**
- Adds remove functionality to status chart 
- Updates legend for xy-chart

### CODE REVIEW

File: packages/dashboards/src/lib/components/timeseries-widget/chart-presets/status-bar-chart/status-bar-chart.component.ts [hunk 1]
```diff
@@ -297,6 +297,19 @@ export class StatusBarChartComponent
                 configuration.gridConfig.sideMargin;
         }
     }
+
+    public displayDeleteButton(): boolean {
+        return !!this.configuration.allowDelete;
```
Comment (stepanstava) [inline packages/dashboards/src/lib/components/timeseries-widget/chart-presets/status-bar-chart/status-bar-chart.component.ts:302]: on what allowDelete depends?

could you use configuration.allowLegendMenu?

please also add a check for TimeseriesWidgetProjectType.PerfstackApp
https://github.com/solarwinds/nova/pull/553/files

**Review by stepanstava — DISMISSED** [review]

**Review by jirkapok — APPROVED** [review]

File: packages/dashboards/src/lib/components/timeseries-widget/chart-presets/status-bar-chart/status-bar-chart.component.less [hunk 2]
```diff
@@ -7,6 +7,7 @@
 
 .status-bar-chart {
     &__legend {
+        position: relative;
```
Comment (findo11) [inline packages/dashboards/src/lib/components/timeseries-widget/chart-presets/status-bar-chart/status-bar-chart.component.less:10]: this probably shouldn't be here

File: packages/dashboards/src/lib/components/timeseries-widget/chart-presets/status-bar-chart/status-bar-chart.component.ts [hunk 3]
```diff
@@ -318,6 +318,23 @@ export class StatusBarChartComponent
...
 
+    public displayDeleteButton(): boolean {
+        return (
+            !!this.configuration.allowLegendMenu &&
+            this.configuration.projectType ===
+                TimeseriesWidgetProjectType.PerfstackApp
+        );
+    }
+
+    public removeMetric(metricId: string): void {
```
Comment (findo11) [inline packages/dashboards/src/lib/components/timeseries-widget/chart-presets/status-bar-chart/status-bar-chart.component.ts:329]: you can move this method to TimeseriesChartComponent and use it from both status and xy.

File: packages/dashboards/src/lib/components/timeseries-widget/chart-presets/xy-chart/xy-chart.component.html [hunk 4]
```diff
@@ -117,16 +117,18 @@
...
+                >
+                    <div class="descriptions">
                         <div
                             *ngIf="!legendSeries.link"
                             class="description description-primary"
                             [ngClass]="{
                                 'nui-text-link-small link': seriesInteractive
                             }"
                             (click)="onPrimaryDescClick($event, legendSeries)"
-                            [title]="legendSeries.legendDescriptionPrimary"
```
Comment (findo11) [inline packages/dashboards/src/lib/components/timeseries-widget/chart-presets/xy-chart/xy-chart.component.html:129]: why did you remove this?

File: packages/dashboards/examples/src/components/test/visual/timeseries/widgets.ts [hunk 5]
```diff
@@ -746,7 +746,13 @@ export const widgetConfigs: IWidget[] = [
                             leftAxisLabel: "Utilization (%)",
                             preset: TimeseriesChartPreset.StatusBar,
                             collectionId: collectionId2,
-                        } as ITimeseriesWidgetConfig,
+                            allowLegendMenu: true,
+                            projectType:
+                                TimeseriesWidgetProjectType.PerfstackApp,
+                            gridConfig: {
+                                fixedLayout: true,
```
Comment (stepanstava) [inline packages/dashboards/examples/src/components/test/visual/timeseries/widgets.ts:753]: please create a new widgets that will have PerfstackApp projectType instead of modifying existing.

**Review by findo11 — APPROVED** [review]
